src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -129,10 +129,18 @@ public void launch(SlaveComputer computer, TaskListener listener) {
             String podId = pod.getMetadata().getName();
             String namespace = StringUtils.defaultIfBlank(slave.getNamespace(), client.getNamespace());
 
-            LOGGER.log(Level.FINE, "Creating Pod: {0} in namespace {1}", new Object[]{podId, namespace});
-            pod = client.pods().inNamespace(namespace).create(pod);
-            LOGGER.log(Level.INFO, "Created Pod: {0} in namespace {1}", new Object[]{podId, namespace});
-            listener.getLogger().printf("Created Pod: %s in namespace %s", podId, namespace);
+            Pod checkPod = client.pods().inNamespace(namespace).withName(podId).get();
```

Comment:
A simple test (https://paste.fedoraproject.org/paste/rwMXdUC670i5PIZFAokjlQ) using the current master gives this in the logs when a restart happens:

https://paste.fedoraproject.org/paste/BRTBZEMvyPxsOo2BJJf6uw

With or without idleMinutes

Reply:
interesting, this was changed in abb2b63b7c0ada20fb2fb0b98337755daae6bc72, can you take a look @vlatombe

Reply:
I guess a simple check of `launched` value at the beginning of the method would be enough. If the pod has been deleted in the mean time we may risk launching another one, leading to an inconsistent state anyway.

Reply:
@Vlatombe so you are ok with the approach in this PR?

Reply:
I think the check pod logic can be replaced by a simple check for `launched==true`. The launcher is meant to be called only once. On restart we just expect the agent to connect back.

Reply:
@Vlatombe  but how is `launched` being persisted across restarts? on a restart, the attempt to reconnect fails with a handshake error:

https://paste.fedoraproject.org/paste/yHv3FoJ35r7vG8J3hWwYEQ

Reply:
@scoheb Should be in the slave's config.xml > launcher > launched.

Dunno about the handshake for now since the real error is likely in the master's logs.

Reply:
I checked the slave config.xml once launched and while running a build:

https://paste.fedoraproject.org/paste/cp~cnrkWdoDkoZS~OLkWLg

Looks like the launched property (when true) is not saved. Do we need a `slave.save()` after at the end of the `launch()` ?

Reply:
Sounds like the right move.

Reply:
@Vlatombe  

With master and ensuring 'launched == true' in config.xml (via direct save() in launch()), here is Jenkins log after restart:
https://paste.fedoraproject.org/paste/UhqADZdp010vXMAk2HI8nw

No trace of why the slave is not connecting properly...